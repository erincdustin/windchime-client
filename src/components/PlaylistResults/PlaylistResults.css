

.results {
  display: block;
  background-color: rgb(243, 243, 243);
  height: 500px;
  width: 100%;
  border-radius: 10px;
  /* margin-left: 0px; */
  text-align: center;
  margin-bottom: 25px;
}

.iframe {
  margin: auto;
  text-align: center;
  width: 250px;
  height: 380px;
}

.btn-results {
  margin-bottom: 15px;
  margin-left: 5px;
  margin-right: 5px;
}

@media screen and (min-width: 900px) {
  .iframe {
    width: 350px;
  }
}

@media screen and (min-width: 1200px) {
  .iframe {
    width: 400px;
  }
}

/* .container {
  position: relative;
  width: 250px;
  height: 80px;
}

.iframe {
  position: absolute;
  top: 0;
  left: 0;
  z-index: 1;
  width: 250px;
  height: 80px;
  /* opacity: 0; */
  /* transition: opacity 700ms; */
/* } */
/* 
.loading {
  margin: auto;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 250px;
  height: 80px;
  z-index: 0;
}

.loading i{
  animation: rotation 2s infinite linear;
}

@keyframes rotation {
  from {
    transform: rotate(0deg);
  } to {
    transform: rotate(359deg);
  }
}  */