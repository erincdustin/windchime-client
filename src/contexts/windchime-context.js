import React, { Component } from 'react'

const WindChimeContext = React.createContext({
    error: '',
    songs: null,
    weather: null,
    genreChoice: null,
    id: null,
    artists: null,
    topArtists: null,
    playlistId: null,
    userPlaylists: null,
    snapshot: null,
    targetEnergy: null,
    targetValence: null,
    targetTempo: null,
    targetPopularity: null,
    setError: () => { },
    clearError: () => { },
    setSongs: () => { },
    setWeather: () => { },
    setGenreChoice: () => { },
    setId: () => { },
    setArtists: () => { },
    setTopArtists: () => { },
    setPlaylistId: () => { },
    setUserPlaylists: () => { },
    setSnapshot: () => { },
    setTargetEnergy: () => { },
    setTargetValence: () => { },
    setTargetTempo: () => { },
    setTargetPopularity: () => { },
})

export default WindChimeContext;

export class WindChimeProvider extends Component {
    constructor(props) {
        super(props)
        const state = {
            error: '',
            songs: null,
            weather: null,
            genreChoice: null,
            id: null,
            artists: null,
            topArtists: null,
            userPlaylists: null,
            playlistId: null,
            snapshot: null,
            targetEnergy: null,
            targetValence: null,
            targetTempo: null,
            targetPopularity: null,
            setError: this.setError,
            clearError: this.clearError,
            setSongs: this.setSongs,
            setWeather: this.setWeather,
            setGenreChoice: this.setGenreChoice,
            setId: this.setId,
            setArtists: this.setArtists,
            setTopArtists: this.setTopArtists,
            setPlaylistId: this.setPlaylistId,
            setUserPlaylists: this.setUserPlaylists,
            setSnapshot: this.setSnapshot,
            setTargetEnergy: this.setTargetEnergy,
            setTargetValence: this.setTargetValence,
            setTargetTempo: this.setTargetTempo,
            setTargetPopularity: this.setTargetPopularity
          }   
        this.state = state; 
    }

    setError = error => {
      console.error(error)
      this.setState({ error })
    }

    clearError = () => {
      this.setState({ error: '' })
    }

    setUserPlaylists = (userPlaylists) => {
      this.setState({ userPlaylists })
    }

    setSongs = (songs) => {
      this.setState({ songs })
    }

    setWeather = (weather) => {
      this.setState({ weather })
    }

    setGenreChoice = (genreChoice) => {
      this.setState({ genreChoice })
    }

    setId = (id) => {
      this.setState({ id })
    }

    setArtists = (artists) => {
      this.setState({ artists })
    }

    setTopArtists = (topArtists) => {
      this.setState({ topArtists })
    }

    setPlaylistId = (playlistId) => {
      this.setState({ playlistId })
    }

    setSnapshot = (snapshot) => {
      this.setState({ snapshot })
    }

    setTargetEnergy = (targetEnergy) => {
      this.setState({ targetEnergy })
    }

    setTargetValence = (targetValence) => {
      this.setState({ targetValence })
    }

    setTargetTempo = (targetTempo) => {
      this.setState({ targetTempo })
    }

    setTargetPopularity = (targetPopularity) => {
      this.setState({ targetPopularity })
    }

    render() {
        return (
            <WindChimeContext.Provider value={this.state}>
                {this.props.children}
            </WindChimeContext.Provider>
        )
    }
}
