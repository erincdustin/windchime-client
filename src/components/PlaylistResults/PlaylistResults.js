import React from 'react';
import {Link} from 'react-router-dom';
import './PlaylistResults.css';

function PlayListResults(props) {
  let styling = '';
  if (props.weather.PrecipitationType === "Rain") {
    styling = ".rain"
  }
  let playlist = '';
  const url =  `https://open.spotify.com/embed/user/spotify/playlist/${props.playlistId}`;
  if(props.snapshot !== null) {
    playlist = 
    <div>
      <h3>Don't like this playlist?</h3>
      <button className="btn-results"><Link className="link" to="/changeGenreParams">Change Playlist Mood Settings</Link></button>
      {/* <button className="btn-results"><Link className="link" to="/genreOption">Change Genre</Link></button> */}
      <iframe class="iframe" title={props.playlistId} src={url} width="250" height="380" frameBorder="0" allowTransparency="true" ></iframe>
    </div>;
  }

  return (
    <div>
      <h3 className="ribbon three">Your Playlist</h3>
      <div className="results">{playlist}</div>
    </div>
  );
}

export default PlayListResults;