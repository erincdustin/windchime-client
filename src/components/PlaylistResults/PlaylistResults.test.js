import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'
import { library } from '@fortawesome/fontawesome-svg-core'
import { faWind } from '@fortawesome/free-solid-svg-icons'
import { WindChimeProvider } from '../../contexts/windchime-context';
import PlaylistResults from './PlaylistResults';

library.add(faWind)

describe(`PlaylistResults`, () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(<WindChimeProvider>
          <BrowserRouter>
            <PlaylistResults />
          </BrowserRouter>
        </WindChimeProvider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
