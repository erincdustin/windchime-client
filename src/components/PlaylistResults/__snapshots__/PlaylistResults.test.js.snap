// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistResults renders as expected 1`] = `
<div>
  <div
    className="ribbon three"
  >
    <span
      className="ribbon-header"
    >
      Your Playlist
    </span>
  </div>
  <div
    className="results"
  >
    <div>
      <button
        className="btn mood"
      >
        <a
          className="link"
          href="/changeGenreParams"
          onClick={[Function]}
        >
          Change Mood Settings
        </a>
      </button>
      <div
        className="loading-result"
      >
        Loading...
      </div>
      <iframe
        allowTransparency="true"
        className="iframe"
        frameBorder="0"
        src="https://open.spotify.com/embed/user/spotify/playlist/12345"
        title="12345"
      />
    </div>
  </div>
</div>
`;
